luigi/worker.py

```diff
@@ -447,6 +455,11 @@ class worker(Config):
     check_unfulfilled_deps = BoolParameter(default=True,
                                            description='If true, check for completeness of '
                                            'dependencies before running a task')
+    check_complete_on_run = BoolParameter(default=False,
+                                          config_path=dict(section='core', name='check-complete-on-run'),
```

Comment:
i don't see the reason to allow this to be included in the legacy `core` section with alt name. I'd recommend just removing the config_path option and force the inclusion of `check_complete_on_run` in a `[worker]` section.

Reply:
could you clarify what you mean by "force the inclusion in a `[worker]` section"? i had assumed that specifying `config_path` was required to read values from the config file, so the solution here would be to just change `section` to `worker`.

Reply:
Currently, this new configuration option could be set as either:
```ini
[worker]
check_complete_on_run: True
```
bc it is a python variable named `check_complete_on_run` and within the `worker` `luigi.Config` class

or
```ini
[core]
check-complete-on-run: True
```
because you've specified a `config_path` which differs from the Config class name and variable name.

---

I joined the Luigi project after Config classes were already introduced. But i understood the `config_path` parameter option was a compatibility utility to transition the dash-delimited naming to underscore-delimited naming, as well as changing the default configuration section where the variable could be set.

So when i say "force the inclusion in the [worker] section", i just mean remove the `config_path=...`. Thus, requiring the variable to be set only from the [worker] section (and not allowing `check-complete-on-run` in the [core]`.